dashboard/app/views/layouts/_school_info_confirmation_dialog.html.haml

```diff
@@ -0,0 +1,25 @@
+-# Rendering this partial causes the school info confirmation dialog to be shown to the
+-# user.
+- require 'country_codes'
+- require 'state_abbr'
+- require 'geocoder'
+
+- location = Geocoder.search(request.ip).try(:first)
+-# geocoder sometimes shows localhost's country as RD/Reserved
+- us_ip = location.nil? || ['US', 'RD'].include?(location.country_code.to_s.upcase)
+- current_user_school_info = current_user.school_info
+
+- script_data = {}
+- script_data[:formUrl] = registration_url('user')
+- script_data[:authTokenName] = request_forgery_protection_token.to_s
+- script_data[:authTokenValue] = form_authenticity_token
+- script_data[:existingSchoolInfo] = {}
+- if current_user_school_info
+  - script_data[:existingSchoolInfo] = current_user_school_info.slice(:id, :country, :school_type, :school_id, :school_name, :full_address)
+- script_data[:existingSchoolInfo][:country] ||= 'United States' if us_ip
+- script_data[:existingSchoolInfo][:school_name] ||= current_user_school_info.school.name
```

Comment:
There's a nontrivial amount of logic (defaults, etc) in how we build up the `script_data` - maybe this is something we can extract and test, and potentially share with _school_info_interstitial.html.haml.

Reply:
I agree!!